pkg/azureutils/azure_disk_utils.go

```diff
@@ -476,11 +478,19 @@ func ParseDiskParameters(parameters map[string]string) (ManagedDiskParameters, e
 			if deviceSettings, err := optimization.GetDeviceSettingFromAttribute(k); err == nil {
 				diskParams.DeviceSettings[filepath.Join(consts.DummyBlockDevicePathLinux, deviceSettings)] = v
 			} else {
-				return diskParams, fmt.Errorf("invalid parameter %s in storage class", k)
+				invalidParameters = append(invalidParameters, k)
 			}
 		}
 	}
-	return diskParams, nil
+
+	if len(invalidParameters) > 0 {
+		err = fmt.Errorf("invalid parameters %s in storage class", strings.Join(invalidParameters, ", "))
```

Comment:
Instead on unconditionally filtering out unknown parameters **and** returning an error, this function should take an `enum` parameter that would specify either return an error on unknown parameters (`strictValidation`) or filter out unknown parameters (`ignoreUnknown`). This way, the intent and functionality is clear at the call point, and other errors can be responded to.